core/descriptors.runtime/src/org/jetbrains/kotlin/descriptors/runtime/components/ReflectKotlinClass.kt

```diff
@@ -243,6 +243,10 @@ private object ReflectClassStructure {
                     componentType == Class::class.java -> for (element in value as Array<*>) {
                         v.visitClassLiteral((element as Class<*>).classLiteralValue())
                     }
+                    Annotation::class.java.isAssignableFrom(componentType) -> for (element in value as Array<*>) {
+                        val vv = v.visitAnnotation(componentType.classId) ?: return
```

Comment:
It seems like `?: continue` would make more sense here rather than `?: return`, don't you think? Because if the visitor is not interested in one annotation instance in the array, we shouldn't disregard the whole array, but rather try to visit the rest too.

At this point it makes no difference, I think, because only the implementation from `BinaryClassAnnotationAndConstantLoaderImpl.kt` can arrive here, and it always returns a visitor. But if this code is refactored/reused somewhere else, it might behave unexpectedly...

Reply:
Yes, you are right. I was thinking about early return since all the implementations returns a visitor except `CollectStringArrayAnnotationVisitor`. It's a mistake.

I'm running tests and will update the PR once it finishes.

Reply:
Tested and updated. Thanks!